dracut/99setup-root/initrd-setup-root

```diff
@@ -7,10 +8,52 @@
 # files if needed and make it easier to debug. In fact, it was required
 # to use first_boot=1 with PXE to get a proper rootfs which is strange.
 
+function usrbin() {
+  LD_LIBRARY_PATH=/sysusr/usr/lib64 /sysusr/usr/bin/"$@"
+}
+
+function walksysroot() {
+  local topdir="$1"
+  shift
+  local action="$1"
+  shift
+  while IFS= read -r -d '' entry; do
+    "${action}" "${entry}" || true
+  done < <(chroot /sysroot /usr/bin/find "${topdir}" -xdev "$@" -print0 | usrbin tac -s "")
+  # Do a chroot to be able to pass "-regex /etc/something" without having to prepend /sysroot
+  # in the regex. Also do the print0 as last action, after filtering.
+  # Use tac to process deeper files/dirs before upper dirs.
+}
+
 # /etc/machine-id after a new image is created:
 COREOS_BLANK_MACHINE_ID="42000000000000000000000000000042"
 MACHINE_ID_FILE="/sysroot/etc/machine-id"
 
+function selectiveosreset() {
+  local entry="/sysroot$1"
+  # Don't remove /sysroot itself
+  [ "${entry}" = "/sysroot" ] && return
+  [ "${entry}" = "/sysroot/" ] && return
+  # Don't remove the active /usr mount point
+  [ "${entry}" = "/sysroot/usr" ] && return
```

Comment:
Should we also have the check for `"/sysroot/usr/"`? Note the trailing slash.

Reply:
Find doesn't add the slash at the end but maybe it's still a good idea